src/runtime/utils/refreshHandler.ts

```diff
@@ -0,0 +1,79 @@
+import type { RefreshHandlerConfig, RefreshHandler } from '../types'
+import { useRuntimeConfig, useAuth, useAuthState } from '#imports'
+
+export const defaultRefreshHandler: RefreshHandler & {
+  config?: RefreshHandlerConfig
+  refetchIntervalTimer?: ReturnType<typeof setInterval>
+  refreshTokenIntervalTimer?: ReturnType<typeof setInterval>
+  visibilityHandler(): void
+} = {
+  // Session configuration keep this for reference
+  config: undefined,
+
+  // Refetch interval
+  refetchIntervalTimer: undefined,
+
+  // TODO: find more Generic method to start a Timer for the Refresh Token
+  // Refetch interval for local/refresh schema
+  refreshTokenIntervalTimer: undefined,
+
+  visibilityHandler () {
+    // Listen for when the page is visible, if the user switches tabs
+    // and makes our tab visible again, re-fetch the session, but only if
+    // this feature is not disabled.
+    if (this.config?.enableRefreshOnWindowFocus && document.visibilityState === 'visible') {
+      useAuth().getSession()
+    }
+  },
+
+  init (config: RefreshHandlerConfig): void {
+    this.config = config
+
+    const runtimeConfig = useRuntimeConfig().public.auth
+
+    const { data } = useAuthState()
+    const { getSession } = useAuth()
+
+    document.addEventListener('visibilitychange', () => {
+      this.visibilityHandler()
+    }, false)
```

Comment:
I think you changed the semantics here: you are adding an anonymous event handler and not really removing it on `destroy`

Reply:
Done. Now using the function ref inside the object